core/src/test/java/io/grpc/internal/SpiffeUtilTest.java

```diff
@@ -193,4 +207,119 @@ public void spiffePathFormatTest() {
           + "underscores ([a-zA-Z0-9.-_])", iae.getMessage());
     }
   }
+
+  public static class CertificateApiTest {
```

Comment:
nit: "CertificateApiTest" is very vague - can we say "ExtractSpiffeIdTest" instead, so that the test name is closely tied to the API we are testing?

I also see that we're grouping this together with the "load SPIFFE bundle APIs" - should we separate them, for consistency with what we did for parsing?

Reply:
Yeah, I couldn't find a good name here. So we test both extract and load APIs, and what unifies them is the fact that we do some operation with certs (extracting spiffe from them, or loading them in a structure).
If we separate them, then we'll need duplicate `setUp()` method to load the same certs for both. To avoid that we might introduce an extra parenting class for both, but it'll be the same question about how to name it.
WDYT?

Reply:
Ack ok, fine leaving it.